metpy/calc/tools.py

```diff
@@ -831,3 +832,296 @@ def _less_or_close(a, value, **kwargs):
+                f[slice2] / (combined_delta * delta[delta_slice1]))
+
+    # Now the "right" edge
+    slice0[axis] = slice(-3, -2)
+    slice1[axis] = slice(-2, -1)
+    slice2[axis] = slice(-1, None)
+    delta_slice0[axis] = slice(-2, -1)
+    delta_slice1[axis] = slice(-1, None)
+
+    combined_delta = delta[delta_slice0] + delta[delta_slice1]
+    right = 2 * (f[slice0] / (combined_delta * delta[delta_slice0]) -
+                 f[slice1] / (delta[delta_slice0] * delta[delta_slice1]) +
+                 f[slice2] / (combined_delta * delta[delta_slice1]))
+
+    return concatenate((left, center, right), axis=axis)
+
+
+def gradient(f, **kwargs):
+    """Calculate the gradient of a grid of values.
+
+    Works for both regularly-spaced data, and grids with varying spacing.
+
+    Either `x` or `deltas` must be specified.
+
+    Parameters
+    ----------
+    f : array-like
+        Array of values of which to calculate the derivative
+    x : array-like, optional
+        The coordinate values corresponding to the grid points in `f`
+    deltas : array-like, optional
+        Spacing between the grid points in `f`. There should be one item less than the size
+        of `f` along `axis`.
+
+    Returns
+    -------
+    array-like
+        The first derivative calculated along each axis in the original array
+
+    See Also
+    --------
+    laplacian
+
+    """
+    pos_kwarg, positions = _process_gradient_args(kwargs)
+    return tuple(first_derivative(f, axis=ind, **{pos_kwarg: positions[ind]})
+                 for ind, pos in enumerate(positions))
+
+
+@exporter.export
+def laplacian(f, **kwargs):
+    """Calculate the laplacian of a grid of values.
+
+    Works for both regularly-spaced data, and grids with varying spacing.
+
+    Either `x` or `deltas` must be specified.
+
+    Parameters
+    ----------
+    f : array-like
+        Array of values of which to calculate the derivative
+    x : array-like, optional
+        The coordinate values corresponding to the grid points in `f`
+    deltas : array-like, optional
+        Spacing between the grid points in `f`. There should be one item less than the size
+        of `f` along `axis`.
+
+    Returns
+    -------
+    array-like
+        The laplacian
+
+    See Also
+    --------
+    gradient
+
+    """
+    pos_kwarg, positions = _process_gradient_args(kwargs)
+    return sum(second_derivative(f, axis=ind, **{pos_kwarg: positions[ind]})
+               for ind, pos in enumerate(positions))
+
+
+def _broadcast_to_axis(arr, axis, ndim):
+    """Handle reshaping coordinate array to have proper dimensionality.
+
+    This puts the values along the specified axis.
+    """
+    if arr.ndim == 1 and arr.ndim < ndim:
+        new_shape = [1] * ndim
+        new_shape[axis] = arr.size
+        arr = arr.reshape(*new_shape)
+    return arr
+
+
+def _process_gradient_args(kwargs):
+    """Handle common processing of arguments for gradient and gradient-like functions."""
```

Comment:
Should there be an exception raised when a user through accident or mis-understanding passes both instead of just grabbing `deltas` and ignoring `x` silently?

Reply:
Fair enough.